shim/opentelemetry-opencensus-shim/src/opentelemetry/shim/opencensus/_shim_tracer.py

```diff
@@ -36,12 +40,57 @@ def get_shim_span_in_context() -> ShimSpan:
     return context.get_value(_SHIM_SPAN_KEY)
 
 
+def set_oc_span_in_context(
+    oc_span_context: SpanContext, ctx: context.Context
+) -> context.Context:
+    """Returns a new OTel context based on ctx with oc_span_context set as the current span"""
+
+    # If no SpanContext is passed to the opencensus.trace.tracer.Tracer, it creates a new one
+    # with a random trace ID and a None span ID to be the parent:
+    # https://github.com/census-instrumentation/opencensus-python/blob/2e08df591b507612b3968be8c2538dedbf8fab37/opencensus/trace/tracer.py#L47.
+    #
+    # OpenTelemetry considers this an invalid SpanContext and will ignore it, so we can just
+    # return early
+    if oc_span_context.span_id is None:
+        return ctx
+
+    trace_id = int(oc_span_context.trace_id, 16)
+    span_id = int(oc_span_context.span_id, 16)
+    is_remote = oc_span_context.from_header
+    trace_flags = (
+        _SAMPLED if oc_span_context.trace_options.get_enabled() else None
+    )
+    trace_state = (
+        trace.TraceState(tuple(oc_span_context.tracestate.items()))
+        # OC SpanContext does not validate this type
+        if isinstance(oc_span_context.tracestate, Tracestate)
+        else None
+    )
+
+    return trace.set_span_in_context(
+        trace.NonRecordingSpan(
+            trace.SpanContext(
+                trace_id=trace_id,
+                span_id=span_id,
+                is_remote=is_remote,
+                trace_flags=trace_flags,
```

Comment:
If you pass in actual `None` values for `trace_flags` and `trace_state`, wouldn't they be actually set to `None` instead of using the defaults?